internal/pkg/service/buffer/api/receive/quota/quota.go

```diff
@@ -24,50 +25,50 @@ const (
 
 type Checker struct {
 	clock         clock.Clock
-	config        config.APIConfig
+	config        config.Config
 	cachedL2Stats *statsCache.L2
 
 	// nextLogAt prevents errors from flooding the log
 	nextLogAtLock *sync.RWMutex
-	nextLogAt     map[key.ReceiverKey]time.Time
+	nextLogAt     map[key.SinkKey]time.Time
 }
 
 type dependencies interface {
-	APIConfig() config.APIConfig
+	Config() config.Config
 	Clock() clock.Clock
 	StatisticsL2Cache() *statsCache.L2
 }
 
 func New(d dependencies) *Checker {
 	return &Checker{
 		clock:         d.Clock(),
-		config:        d.APIConfig(),
+		config:        d.Config(),
 		cachedL2Stats: d.StatisticsL2Cache(),
 		nextLogAtLock: &sync.RWMutex{},
-		nextLogAt:     make(map[key.ReceiverKey]time.Time),
+		nextLogAt:     make(map[key.SinkKey]time.Time),
 	}
 }
 
-// Check checks whether the size of records that one receiver can buffer in etcd has not been exceeded.
-func (c *Checker) Check(ctx context.Context, k key.ReceiverKey) error {
-	stats, err := c.cachedL2Stats.ReceiverStats(ctx, k)
+// Check checks whether the size of records that one sink can buffer has not been exceeded.
+// The method compares used disk space, on all disks, with the provided quota value.
+func (c *Checker) Check(ctx context.Context, k key.SinkKey, quota datasize.ByteSize) error {
+	stats, err := c.cachedL2Stats.SinkStats(ctx, k)
```

Comment:
Update types/naming:
- `Receiver` -> `Source`
- `Export` -> `Sink`

We intendently  do not control the quota per `Source` (`Receiver`), but at a lower level - per `Sink` (`Export`). In this way, we can tell more precisely where the problem is.

I added `quota datasize.ByteSize` argument, so the responsibility for loading the quote value has been moved to the upper code.